### ISSUE TITLE
Mysterious log spam

### ISSUE THREAD

**Issue by ozin370** [KSP-KOS/KOS#2084]
This just started spamming my log many times per second while running a script. This was after reloading scenes 3-4 times previously in the session, where I didn't notice it occurring (running the same script).

```[EXC 23:32:02.299] ArgumentOutOfRangeException: Argument is out of range.
Parameter name: index
	System.Collections.Generic.List`1[kOS.Safe.Screen.IScreenBufferLine].get_Item (Int32 index)
	kOS.Screen.TermWindow.TerminalGui (Int32 windowId)
	UnityEngine.GUI.CallWindowDelegate (UnityEngine.WindowFunction func, Int32 id, UnityEngine.GUISkin _skin, Int32 forceRect, Single width, Single height, UnityEngine.GUIStyle style)
[ERR 23:32:02.301] GUI Error: You are pushing more GUIClips than you are popping. Make sure they are balanced)```

**Comment by hvacengi** [comment]
Do you adjust the screen size in your script at all?  It could be a disconnect between the number of lines in the buffer and the number of lines the UI wants to display.

**Comment by Dunbaratu** [comment]
There is a pattern used in the unity immediate GUI where you group things in 'horizontal groups' or 'vertical groups', by pairing begin/end statements (i.e. `BeginHorizontal` and `EndHorizontal`).  If the GUI code doesn't match these up properly so the begins don't match the ends, you get this error message.

It could be a bug in kOS, but I'm not sure how it would trigger.

**Comment by ozin370** [comment]
Yeah, I do set the terminal height and width. When the ship stops being the active vessel I shrink the terminal size significantly.

It doesn't seem to trigger often though, but I will try to keep an eye out for it.
